app/services/redux.js

```diff
@@ -1,39 +1 @@
-import Ember from 'ember';
-import redux from 'redux';
-import reducers from '../reducers/index';
-import enhancers from '../enhancers/index';
-import middlewareConfig from '../middleware/index';
-
-const { assert, isArray } = Ember;
-
-// Handle "classic" middleware exports (i.e. an array), as well as the hash option
-const extractMiddlewareConfig = (mc) => {
-  assert(
-    'Middleware must either be an array, or a hash containing a `middleware` property',
-    isArray(mc) || mc.middleware
-  );
-  return isArray(mc) ? { middleware: mc } : mc;
-}
-
-// Destructure the middleware array and the setup thunk into two different variables
-const { middleware, setup = () => {} } = extractMiddlewareConfig(middlewareConfig);
-
-var { createStore, applyMiddleware, combineReducers, compose } = redux;
-var createStoreWithMiddleware = compose(applyMiddleware(...middleware), enhancers)(createStore);
-
-export default Ember.Service.extend({
-  init() {
-    this.store = createStoreWithMiddleware(combineReducers(reducers));
-    setup(this.store);
-    this._super(...arguments);
-  },
-  getState() {
-    return this.store.getState();
-  },
-  dispatch(action) {
-    return this.store.dispatch(action);
-  },
-  subscribe(func) {
-    return this.store.subscribe(func);
-  }
-});
+export {default} from 'ember-redux/services/redux';
```

Comment:
In other words, we'll still need to support pull in all the user's reducers, middlewares, and enhancers in the `app` directory's redux service, which probably means we might need to refactor our `addon/services/redux` a bit to be more accommodating to this.

Consider my implementation here: https://github.com/foxnewsnetwork/ember-redux/blob/feature/addonify-redux-service/addon/services/redux.js#L30

```javascript
import reducers from '../reducers/optional';
import enhancers from '../enhancers/index';
import middlewareConfig from '../middleware/index';

export default Ember.Service.extend({
  middlewareConfig,
  reducers,
  enhancers,
  makeStoreInstance,
  init() {
    // gets the middlewareConfig from the instance of this service
    // if you'd like to customize your middlewares, you'd probably do it here
    const middlewareConfig = get(this, 'middlewareConfig');
    const reducers = get(this, 'reducers');
    const enhancers = get(this, 'enhancers');

    this.store = this.makeStoreInstance({ middlewareConfig, reducers, enhancers });
    this._super(...arguments);
  }
...otherstuff
}
``` 

which would allow us in `app/services/redux` to just do:
```javascript
import ReduxService from 'ember-redux/services/redux';
import reducers from '../reducers/index';
import enhancers from '../enhancers/index';
import middlewareConfig from '../middleware/index';

export default ReduxService.extend({ reducers, enhancers, middlewareConfig });
```
https://github.com/foxnewsnetwork/ember-redux/blob/feature/addonify-redux-service/app/services/redux.js